### ISSUE TITLE
Ability to hide posts manually

### ISSUE THREAD

**Issue by aproposnix** [LemmyNet/lemmy#1403]
### Describe the solution you'd like

As a user, I would like to have the ability to manually hide posts from showing up on my feed. This is the same functionality available on Reddit. 

### Describe alternatives you've considered

There is no way to do this AFAIK

There is another request for hiding posts but it's about automatically hiding read posts which is NOT the same as what I am requesting. https://github.com/LemmyNet/lemmy/issues/1072

**Comment by dessalines** [comment]
This wouldn't be too difficult to add, but I'd also wanna do it after #1072

**Comment by dessalines** [comment]
I'm wondering how necessary this is. Hiding a single post has limited usefulness, since it'd be gone from your feed within a few days anyway. 

Also lemmy already supports:

- Blocking communities and users.
- Hiding read posts.

Which will do a much better job than manually hiding a single post.

**Comment by justincmoy** [comment]
Is it worth reconsidering adding this to Lemmy? It seems like it would be better to add this into the platform than to have one-off implementations in each client: https://github.com/buresdv/Mlem/issues/36

**Comment by Nutomic** [comment]
I believe this can already be done by marking the post as read.

**Comment by acosmichippo** [comment]
> I believe this can already be done by marking the post as read.

personally I like leaving read posts visible in case I want to revisit the comments at a later time.  but some posts I am not interested in at all, and would prefer them not showing up at the top of my front page for multiple days.  but maybe I'm just weird.

**Comment by ndcontini** [comment]
I'd like to emphasize that I'd like to this feature. This is *really* useful when you just really don't want to see a post in your feed anymore. A good example is if someone asks a question in a programming community, and you've already answered the question. I only need to revisit that post if someone responds to my comment, which will give me a notification linking me back to the post. Another example is if I just *really* don't like some sort of fan art or topic and just want to hide the post. This is basically a good alternative to being a hater in the comments; it's a lot easier to resist the temptation if you just hide the post from view.

**Comment by waldyrious** [comment]
I'm in the exact situation as @acosmichippo — I often want to hide certain posts while keeping the read ones visible.

> Is it worth reconsidering adding this to Lemmy? It seems like it would be better to add this into the platform than to have one-off implementations in each client: https://github.com/buresdv/Mlem/issues/36

Agreed — here's another issue asking for this feature in a different client (Sync): https://github.com/laurencedawson/sync-for-lemmy/issues/60.

**Comment by waldyrious** [comment]
> Hiding a single post has limited usefulness, since it'd be gone from your feed within a few days anyway.

Just adding a different perspective to this: when using the "Subscribed" view rather than the "Everything" view, and depending on the size and activity level of the communities one is subscribed to, posts may take quite a while to disappear from the feed. In that case, the usefulness of hiding posts becomes more relevant.

**Comment by dessalines** [comment]
I spose I do see the distinction between hiding read posts, and marking a post as hidden:

The first is often a passive action, where by just going to the post, you don't want to see it again. The second is explicit, and requires a button press for that post.

**Comment by GeckoEidechse** [comment]
> I spose I do see the distinction between hiding read posts, and marking a post as hidden:

100%. Hiding a post is a signal that indicates "I don't want to see this" while marking a post as read to hide it indicates "I don't want to see this **because I already saw it**".

They are essentially almost opposites when it comes to gauging user sentiment towards a post where one indicates positive sentiment ("I've already seen this") and one indicates negative sentiment ("I don't want to see this").

This for example is why YouTube asks you for a reason when choose to hide a video from your feed. Of course to my knowledge there's no plan for a per-user personalised recommendation engine for lemmy but the distinction between the two cases is still something to keep in mind ^^

&nbsp;

And to give another use case for manually hiding posts. Imagine being part of a community and you see a content that is disturbing based on your personal believes/experience. Say you are afraid kiwis but still want to participate in a cooking community. By manually hiding a post, you can quickly remove individual posts that might be distressing to you without blocking an entire community.

Of course this could then be coupled with additional approaches such as keyword based filtering etc but that's kinda outside the scope of this issue :P

**Comment by Undertakerrip** [comment]
Please take #3157 into account as it was closed as dupes but it contains additional motivation. In a nutshell: Lemmy does not support hiding already seen pinned posts in a good way. This feature request would address this
